AdminUi/test/AdminUi.Tests.Integration/StepDefinitions/ClientsStepDefinitions.cs

```diff
@@ -57,25 +61,78 @@ public async Task WhenADeleteRequestIsSentToTheClientsEndpoint()
     [When(@"a GET request is sent to the /Clients endpoint")]
     public async Task WhenAGetRequestIsSentToTheClientsEndpoint()
     {
-        _response = await _clientsApi.GetAllClients(_requestConfiguration);
-        _response.Should().NotBeNull();
-        _response.Content.Should().NotBeNull();
+        _clientsResponse = await _clientsApi.GetAllClients(_requestConfiguration);
+        _clientsResponse.Should().NotBeNull();
+        _clientsResponse.Content.Should().NotBeNull();
+    }
+
+    [When(@"a PATCH request is sent to the /Clients/{c.ClientId}/ChangeSecret endpoint")]
+    public async Task WhenAPatchRequestIsSentToTheClientsChangeSecretEndpoint()
+    {
+        _clientSecret = "new-client-secret";
+
+        var changeClientSecretRequest = new ChangeClientSecretRequest()
+        {
+            NewSecret = _clientSecret
+        };
+
+        var requestConfiguration = _requestConfiguration.Clone();
+        requestConfiguration.ContentType = "application/json";
+        requestConfiguration.SetContent(changeClientSecretRequest);
+
+        _clientResponse = await _clientsApi.ChangeClientSecret(_clientId, requestConfiguration);
+
+        _clientResponse.Should().NotBeNull();
+        _clientResponse.Content.Should().NotBeNull();
+    }
+
+    [When(@"a PATCH request is sent to the /Clients/{clientId}/ChangeSecret endpoint")]
+    public async Task WhenAPatchRequestIsSentToTheClientsChangeSecretEndpointForAnInexistentClient()
+    {
+        var changeClientSecretRequest = new ChangeClientSecretRequest()
+        {
+            NewSecret = "new-client-secret"
+        };
+
+        var requestConfiguration = _requestConfiguration.Clone();
+        requestConfiguration.ContentType = "application/json";
+        requestConfiguration.SetContent(changeClientSecretRequest);
+
+        _clientResponse = await _clientsApi.ChangeClientSecret("inexistentClientId", requestConfiguration);
+
+        _clientResponse.Should().NotBeNull();
+        _clientResponse.Content.Should().NotBeNull();
     }
 
     [Then(@"the response contains a paginated list of Clients")]
     public void ThenTheResponseContainsAListOfClients()
     {
-        _response!.Content.Result.Should().NotBeNull();
-        _response!.Content.Result.Should().NotBeEmpty();
-        _response.AssertContentCompliesWithSchema();
+        _clientsResponse!.Content.Result.Should().NotBeNull();
+        _clientsResponse!.Content.Result.Should().NotBeEmpty();
+        _clientsResponse.AssertContentCompliesWithSchema();
+    }
+
+    [Then(@"the response contains Client c with the new client secret")]
+    public void ThenTheResponseContainsAClientWithNewSecret()
```

Comment:
The method name and step description state that you check for the new secret. But the implementation says something else